src/sentry/sentry_metrics/multiprocess.py

```diff
@@ -368,6 +398,28 @@ def process_messages(
             org_id = message["org_id"]
             tags = message.get("tags", {})
 
+            if not valid_metric_name(metric_name):
+                logger.error(
+                    "process_messages.invalid_metric_name",
+                    extra={"org_id": org_id, "metric_name": metric_name},
+                )
+                return []
```

Comment:
istm this would drop the entire batch instead of just the invalid bucket?

Reply:
ah yeah good point, i should update the test to actually test that too